### DISCUSSION TITLE
Increasing CPU Resources for Chartbrew

### DISCUSSION THREAD

**Discussion by alb3rtod** [chartbrew/chartbrew#241] [Q&A] [answered]
Is there a way to allocate more CPU resources for Chartbrew? I have a host with 12 CPUs, but it seems like Chartbrew is only using 1 according to the Docker stats information.

**Comment by razvanilin** [comment, marked as answer]
Hey @alb3rtod, I personally run the managed version at chartbrew.com using the [pm2 cluster mode](https://pm2.keymetrics.io/docs/usage/cluster-mode/). That essentially runs multiple copies of the backend so if one user is using one of the CPUs for an intensive task, the application can still run fine for others. I haven't tried using this with Docker though

If you container can run on multiple CPUs, maybe you can give this a try: https://pm2.keymetrics.io/docs/usage/docker-pm2-nodejs/

**Comment by alb3rtod** [comment]
Hi @razvanilin , I was able to implement PM2 inside the container, and now there is a server instance for each of the host's CPUs. I am currently testing the full results. Do you have any other recommendations to improve Chartbrew performance?

**Reply by alb3rtod** [reply]
I'm using version 3.6.0, by the way.

**Reply by razvanilin** [reply]
The latest version of Chartbrew uses queues and redis to schedule chart and dashboard updates with workers. This will can also improve the overall performance of Chartbrew since the updates don't use one of your node instances in the cluster.

To update to the latest version, you should have redis up and running on your server and set up the new `CB_REDIS_*` variables: https://docs.chartbrew.com/#environmental-variables

**Comment by alb3rtod** [comment]
Thanks, @razvanilin . It seems that the Redis server indeed improves the chart updates in some cases."
